src/build-defs.cmake

```diff
@@ -3,6 +3,12 @@ if(NOT USE_DEFAULT_VISIBILITY)
   set(CMAKE_C_VISIBILITY_PRESET "hidden")
 endif()
 
+# postgres CFLAGS includes -Wdeclaration-after-statement which leads
+# to problems when compiling with -Werror since we aim for C99 and allow
+# that so we strip this flag from PG_CFLAGS before adding postgres flags
+# to our own
+string(REPLACE " -Wdeclaration-after-statement" "" PG_CFLAGS "${PG_CFLAGS}")
```

Comment:
This assumes that there is a space before the flag. Since several spaces does not affect the validity of the flags, I think you can just replace the string without spaces.